src/components/PeopleTable.tsx

```diff
@@ -1,4 +1,172 @@
-export const PeopleTable = () => {
+/* eslint-disable no-else-return */
+/* eslint-disable @typescript-eslint/no-unused-expressions */
+import { useParams, useSearchParams } from 'react-router-dom';
+import { useEffect, useState } from 'react';
+import cn from 'classnames';
+import { Person } from '../types';
+import { PersonItem } from './PersonItem/PersonItem';
+import { SearchLink } from './SearchLink';
+
+type Props = {
+  people: Person[],
+  allParams: string,
+};
+
+enum Sort {
+  notSorted,
+  sortAsc,
+  sortDesc,
+}
+
+type SortBy = 'name' | 'sex' | 'born' | 'died' | null;
+
+export const PeopleTable = ({ people, allParams }:Props) => {
+  const params = useParams();
+  const [slugUser, setSlugUser] = useState<string | undefined>(params.slug);
+  const [searchParams, setSearchParams] = useSearchParams(
+    (window.location.hash).substring(window.location.hash.indexOf('?')),
+  );
+  const [sexSort, setSexSort] = useState<Sort>(Sort.notSorted);
+  const [nameSort, setNameSort] = useState<Sort>(Sort.notSorted);
+  const [bornSort, setBornSort] = useState<Sort>(Sort.notSorted);
+  const [diedSort, setDiedSort] = useState<Sort>(Sort.notSorted);
+  const [sortBy, setSortBy] = useState<SortBy | null>(null);
+
+  const handleSlugUser = (value: string) => {
+    setSlugUser(value);
+  };
+
+  const ActiveLinkSexClassName = (sort: Sort) => cn(
+    { 'fas fa-sort': sort === Sort.notSorted },
+    { 'fas fa-sort-up': sort === Sort.sortAsc },
+    { 'fas fa-sort-down': sort === Sort.sortDesc },
+  );
+
+  const handleSort = (
+    sort: Sort,
+    setSort: (sort: Sort) => void,
+    columnName: string,
+  ) => {
+    if (sort === Sort.notSorted) {
+      setSort(Sort.sortAsc);
+      searchParams.set('sort', columnName);
+      setSearchParams(searchParams);
+    } else if (sort === Sort.sortAsc) {
+      setSort(Sort.sortDesc);
+      searchParams.set('order', 'desc');
+      setSearchParams(searchParams);
+    } else {
+      setSort(Sort.notSorted);
+      searchParams.delete('sort');
+      searchParams.delete('order');
+    }
```

Comment:
Use [`switch case`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/switch) in cases where you constantly compare one value with others